### PR TITLE
Adds search endpoint with a minimal integration test.

### PR DESCRIPTION
**Pull request by dbernstein**
**Initial starting point for simple search implementation **
* * *

**JIRA Ticket**: Resolves: https://jira.lyrasis.org/browse/FCREPO-3282 

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
* Adds two new modules: fcrepo-search-api and fcrepo-search-impl
* Adds an integration test to ensure that the search api is returning a valid json result.

# How should this be tested?
```
java  -Dfcrepo.ocfl.staging.dir=target/fcrepo-data/staging -Dfcrepo.ocfl.storage.root.dir=target/fcrepo-data/ocfl-root -Dfcrepo.ocfl.work.dir=target/fcrepo-data/work    -jar fcrepo-webapp/target/fcrepo-webapp-6.0.0-SNAPSHOT-jetty-console.jar --headless
```

```
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/fcr:search -i
```

Verify that it returns 200 and some json.  Ignore the values within the search results since they are not actually connected to anything yet.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers

### CODE REVIEW

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraSearchIT.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            final ObjectMapper objectMapper = new ObjectMapper();
+            final SearchResult result = objectMapper.readValue(response.getEntity().getContent(), SearchResult.class);
+            assertNotNull(result);
+            assertNotNull(result.getPagination());
+            assertEquals(0, result.getResults().size());
+        }
+    }
+
+    private String encode(final String value) throws Exception {
+        return URLEncoder.encode(value, "UTF-8");
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraSearchIT.java:55]: I believe the recommendation is to use the constant: `StandardCharsets.UTF_8`.

File: fcrepo-search-api/src/main/java/org/fcrepo/search/api/SearchResult.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    @JsonProperty
+    private PaginationInfo pagination = new PaginationInfo();
+    @JsonProperty
+    private List<Map<String, Object>> results = new ArrayList();
+
+    public SearchResult() {
+
+    }
+
+    public PaginationInfo getPagination() {
```
Comment (awoods) [inline fcrepo-search-api/src/main/java/org/fcrepo/search/api/SearchResult.java:39]: I am surprised that the build did not fail, given the lack of JavaDoc on these `public` methods... something must have regressed in our checkstyle configuration.

File: fcrepo-search-api/src/main/java/org/fcrepo/search/api/SearchParameters.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.search.api;
+
+/**
+ * A pojo encapsulating the parameters of a search
+ */
+public class SearchParameters {
+    public SearchParameters(final String query) {
```
Comment (awoods) [inline fcrepo-search-api/src/main/java/org/fcrepo/search/api/SearchParameters.java:24]: JavaDocs explaining something about the nature of the expected `query` parameter will be helpful for whoever comes along to implement this.

File: fcrepo-search-impl/pom.xml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-context</artifactId>
```
Comment (awoods) [inline fcrepo-search-impl/pom.xml:57]: Maybe add the `scope:test` if this is test gear.

File: fcrepo-http-api/pom.xml [hunk 5]
```diff
@@ -58,6 +59,18 @@
...
+    <dependency>
+      <groupId>org.fcrepo</groupId>
+      <artifactId>fcrepo-search-api</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.fcrepo</groupId>
+      <artifactId>fcrepo-search-impl</artifactId>
+      <version>${project.version}</version>
```
Comment (awoods) [inline fcrepo-http-api/pom.xml:71]: I believe this should be a `scope:test` dependency.

File: fcrepo-search-impl/pom.xml (hunk 4 above)
Comment (dbernstein) [reply fcrepo-search-impl/pom.xml:57]: This is a compile time dependency:  required for @Component in SearchServiceImpl.java

File: fcrepo-http-api/pom.xml (hunk 5 above)
Comment (dbernstein) [reply fcrepo-http-api/pom.xml:71]: 👍

File: fcrepo-search-impl/pom.xml (hunk 4 above)
Comment (awoods) [reply fcrepo-search-impl/pom.xml:57]: Agreed. Can you move the dependency above the `<!-- test gear -->` section?
https://github.com/fcrepo4/fcrepo4/pull/1680/files/407b021978a2ef071d1a3aeca7ecd5974d63bc22#diff-8efde854920053f9dd8900c22bdd35d3R44

File: fcrepo-search-api/src/main/java/org/fcrepo/search/api/SearchResult.java (hunk 2 above)
Comment (dbernstein) [reply fcrepo-search-api/src/main/java/org/fcrepo/search/api/SearchResult.java:39]: Me too.  I'm not sure what would have caused the change.

File: fcrepo-search-impl/pom.xml (hunk 4 above)
Comment (awoods) [reply fcrepo-search-impl/pom.xml:57]: It is minor, but would you rather leave this dependency in the `test gear` section?

**Pull request merged**